### PR TITLE
General: update WordPress version requirements to WP 5.6

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

WordPress 5.7 will be released on March 9, 2021, and from that point on Jetpack will require WP 5.6.

This PR introduces the following changes:

- Update our "tested up to" version
- Update the version of WordPress required for Jetpack from WP 5.5 to WP 5.6
- Changes to Toolbar components ([Reference](https://make.wordpress.org/core/2020/11/18/changes-to-toolbar-components-in-wordpress-5-6/))
- Move away from `IconButton` ([Reference](https://make.wordpress.org/core/2020/02/13/an-updated-button-component-in-wordpress-5-4/))
- Move from `formattingControls` to `allowedFormats` ([Reference](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/CHANGELOG.md#300-2019-08-05))
- Other block deprecations

#### Jetpack product discussion

Primary issue: #18842
#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Do the tests pass?
* Test all the blocks that were updated in this PR (see labels), and ensure they still work. You'll want to pay extra attention to the block toolbars that appear above each block.

#### Proposed changelog entry for your changes:

* General: Jetpack now requires WordPress 5.6, and is fully compatible with WordPress 5.7.

### CODE REVIEW

**Review by anomiex — COMMENTED** [review]
Didn't check the JS changes, but seems ok at a glance.

I found the following todo comments that look like they can be done now:

* [projects/packages/sync/src/modules/class-posts.php:578](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/packages/sync/src/modules/class-posts.php#L578): `` * @todo: remove full if statement when WordPress 5.6 is the minimum required version.``
* [projects/packages/lazy-images/tests/php/test_class.lazy-images.php:222](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/packages/lazy-images/tests/php/test_class.lazy-images.php#L222): ``// @todo Remove when WordPress 5.6 is the minimum supported version.``
* [projects/plugins/jetpack/extensions/blocks/anchor-fm/anchor-fm.php:71](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/extensions/blocks/anchor-fm/anchor-fm.php#L71): ``// TODO: Replace `$current_screen->is_block_editor()` with `wp_should_load_block_editor_scripts_and_styles()` that is introduced in WP 5.6.``
* [projects/plugins/jetpack/extensions/blocks/eventbrite/edit.js:173](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/extensions/blocks/eventbrite/edit.js#L173): ``// @todo Remove isDefault and isLarge from Button when the minimum WP version supported by JP uses Gutenberg > 7.2``
* [projects/plugins/jetpack/extensions/blocks/publicize/store/selectors.js:182](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/extensions/blocks/publicize/store/selectors.js#L182): ``// @todo This is a fallback definition, it can be removed when WordPress 5.6 is the minimum supported version.``
* [projects/plugins/jetpack/extensions/blocks/gathering-tweetstorms/editor.js:37](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/extensions/blocks/gathering-tweetstorms/editor.js#L37): ``// @todo The 'core-embed/twitter' check can be removed when WordPress 5.6 is the minimum version.``
* [projects/plugins/jetpack/extensions/blocks/gathering-tweetstorms/editor.js:54](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/extensions/blocks/gathering-tweetstorms/editor.js#L54): ``// @todo The core/embed' check can be removed when WordPress 5.6 is the minimum version.``
* [projects/plugins/jetpack/tests/php/_inc/lib/test-class.jetpack-tweetstorm-helper.php:284](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/tests/php/_inc/lib/test-class.jetpack-tweetstorm-helper.php#L284): ``// @todo This is a fallback definition, it can be removed when WordPress 5.6 is the minimum supported version.``
* [projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php:26](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php#L26): ``// TODO: Replace `$current_screen->is_block_editor()` with `wp_should_load_block_editor_scripts_and_styles()` that is introduced in WP 5.6.``
* [projects/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php:323](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php#L323): ``// @todo This is a fallback definition, it can be removed when WordPress 5.6 is the minimum supported version.``
* [projects/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php:1203](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php#L1203): ``// @todo This fallback can be removed when WordPress 5.6 is the minimum supported version.``
* [projects/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php:1229](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php#L1229): ``// @todo This fallback can be removed when WordPress 5.6 is the minimum supported version.``

Also possibly these, it looks like the PHP version bump it was referring to was moving away from 5.2:

* [projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.tweet.php:8](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.tweet.php#L8): ``// @todo Remove this when WP's PHP version bumps.``
* [projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.hulu.php:45](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.hulu.php#L45): ``// @todo Remove this when WP's PHP version bumps.``
* [projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.ted.php:11](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.ted.php#L11): ``// @todo Remove this when WP's PHP version bumps.``
* [projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.getty.php:45](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.getty.php#L45): ``// @todo Remove this when WP's PHP version bumps.``
* [projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.instagram.php:24](https://github.com/Automattic/jetpack/blob/c4dacfa73cc8e4436ac0892c5a7132f31688e164/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.instagram.php#L24): ``// @todo Remove this when WP's PHP version bumps.``

**Review by anomiex — COMMENTED** [review]
There's a failing JS test.

**Review by anomiex — APPROVED** [review]
Seems sensible. Haven't manually tested though.

**Comment by anomiex** [comment]
JS tests are failing again.

**Comment by jeherve** [comment]
> JS tests are failing again.

I attempted to fix this (see 4 commits above). Unfortunately it will require a larger set of changes that I don't think are appropriate at this point of the release cycle (update the `@wordpress/components` package so the tests can pass). This should happen in #16763. Until then, I've reverted my change and I think we can live with the deprecation notice in the block editor in this case.

**Review by mmtr — COMMENTED** [review]
Good refactor, @jeherve! Reviewed the whole code and tested all affected blocks without noting any regression, except for the Anchor extension which should be easy to fix.

File: projects/plugins/jetpack/extensions/blocks/anchor-fm/anchor-fm.php [hunk 1]
```diff
@@ -67,13 +67,8 @@ function process_anchor_params() {
...
 
-	$current_screen = \get_current_screen();
-	// TODO: Replace `$current_screen->is_block_editor()` with `wp_should_load_block_editor_scripts_and_styles()` that is introduced in WP 5.6.
-	if (
-		$current_screen instanceof \WP_Screen
-		&& ! $current_screen->is_block_editor()
-	) {
-		// Return early if we are not in the block editor.
+	// Return early if we are not in the block editor.
+	if ( wp_should_load_block_editor_scripts_and_styles() ) {
```
[the remark this answers is not in the document]
Comment (jeherve) [reply projects/plugins/jetpack/extensions/blocks/anchor-fm/anchor-fm.php:71]: Ah, great catch!

**Review by mmtr — APPROVED** [review]
LGTM! 🚀

**Comment by jeherve** [comment]
r221664-wpcom

**Pull request merged**